### PR TITLE
Timeline Cleanup

### PR DESCRIPTION
**Pull request by NancyMcCallB**
Clean up TimelineComponent and its test coverage.
Refactor DefaultViewOverlay and fix bugs.

### CODE REVIEW

File: common/api/summary/ui-framework.exports.csv [hunk 1]
```diff
@@ -678,6 +679,8 @@ beta;UserSettingsProvider
 public;UserSettingsStorage 
 internal;useSavedFrontstageState(frontstageDef: FrontstageDef): void
 internal;useSaveFrontstageSettings(frontstageDef: FrontstageDef): void
+public;useScheduleAnimationDataProvider(viewport: ScreenViewport | undefined): ScheduleAnimationTimelineDataProvider | undefined
+beta;useSolarDataProvider(viewport: ScreenViewport | undefined): SolarDataProvider | undefined
```
Comment (pmconne) [inline common/api/summary/ui-framework.exports.csv:683]: Why is the solar one beta?

**Review by pmconne — CHANGES_REQUESTED** [review]
The timeline controls are hugely useful, If they are being made public please advertise that loudly in NextVersion.md.

File: test-apps/ui-test-app/src/frontend/appui/statusfields/DisplayStyleField.tsx [hunk 2]
```diff
@@ -86,12 +86,16 @@ export class DisplayStyleField extends React.Component<StatusFieldProps, Display
...
 
-  private _handleDisplayStyleSelected = (event: React.ChangeEvent<HTMLSelectElement>): void => {
+  private _handleDisplayStyleSelected = async (event: React.ChangeEvent<HTMLSelectElement>) => {
     if (!this.state.viewport)
       return;
 
     const viewport = this.state.viewport;
-    viewport.displayStyle = this.state.displayStyles.get(event.target.value)!.clone();
+    const style = this.state.displayStyles.get(event.target.value)!.clone();
+    if (style)
```
Comment (bsteinbk) [inline test-apps/ui-test-app/src/frontend/appui/statusfields/DisplayStyleField.tsx:95]: We should probably do the processing in an if style { } block.

**Review by bsteinbk — APPROVED** [review]

File: common/api/summary/ui-framework.exports.csv (hunk 1 above)
Comment (NancyMcCallB) [reply common/api/summary/ui-framework.exports.csv:683]: We're not ready with the solar one yet -- we know of at least one problem that requires a refactor. Bill and I are discussing what to do with it.

File: test-apps/ui-test-app/src/frontend/appui/statusfields/DisplayStyleField.tsx (hunk 2 above)
Comment (NancyMcCallB) [reply test-apps/ui-test-app/src/frontend/appui/statusfields/DisplayStyleField.tsx]: Done.

File: docs/changehistory/NextVersion.md [hunk 3]
```diff
@@ -54,3 +54,7 @@ export class ExtensionUiItemsProvider implements UiItemsProvider {
...
 
 - The `version` prop passed to [FrameworkVersion]($ui-framework) component will update the [UiFramework.uiVersion] if necessary keeping the redux state matching the value defined by the prop.
+
+- The [ScheduleAnimationProvider]($ui-framework) is published for use by AppUi apps. Specifying this data provider to a [TimelineComponent]($ui-components) allows animation of the [RenderSchedule.Script]($common) if one exists for the view. A component that automatically detects a schedule script and attaches the data provider to its TimelineComponent can be found in the [DefaultViewOverlay]($ui-framework).
+
+- The [AnalysisAnimationProvider]($ui-framework) is published for use by AppUi apps. Specifying this data provider to a TimelineComponent allows animation of the information in the AnalysisDisplayProperties if the view's [DisplayStyleState]($frontend) contains one. A component that automatically detects analysis data and attaches the data provider to its TimelineComponent can be found in the [DefaultViewOverlay]($ui-framework).
```
Comment (pmconne) [inline docs/changehistory/NextVersion.md:60]: `/s/AnalysisDisplayProperties/AnalysisStyle`.
I have a PR in progress that will promote this class from alpha to public - I'll try to remember to update this to link to the docs once that's completed.

**Review by pmconne — CHANGES_REQUESTED** [review]

File: docs/changehistory/NextVersion.md (hunk 3 above)
Comment (NancyMcCallB) [reply docs/changehistory/NextVersion.md:60]: I'm doing some other work on the TimelineComponent. I'll try to remember to check when I get ready to PR it.

**Review by pmconne — APPROVED** [review]

File: docs/changehistory/NextVersion.md (hunk 3 above)
Comment (pmconne) [reply docs/changehistory/NextVersion.md:60]: There's no such thing as AnalysisDisplayProperties. It doesn't matter - I will change it in my PR after I merge yours.

**Pull request merged**